### PR TITLE
feature(social-authentication): add social authentication with Facebo…

### PR DESCRIPTION
**Pull request by kelvinndmo**
**What does this PR do?**
- Enable users to log in using Facebook, Twitter, and Google

**Description of Tasks to be completed**
- Enable users to signup and log in with Facebook, Twitter, and Google
- Unit tests for Facebook, Twitter and Google authentication

**How should this be manually tested?**
- Clone this [repository](https://github.com/andela/ah-olympians.git) then navigate to `ft-social-auth-164047062`
- Set the environment and install the dependencies by running `pip install -r requirements.txt`
- Setup the database and run migrations.
- You can the Test the application using postman as shown below
Get token from:
[Facebook Developer](https://developers.facebook.com/apps)
[Twitter Developer](https://developer.twitter.com/en/apps)
[Google Developer](https://developers.google.com/oauthplayground)

[Google Developer](https://developers.google.com/oauthplayground)

```
{
	"provider": "google-oauth2",
     "access_token": ""
}
```

google

[Twitter Developer](https://developer.twitter.com/en/apps)
```
{ 
   "provider":"twitter",
   "access_token":"",
   "access_token_secret":""
}
```
twitter

[Facebook Developer](https://developers.facebook.com/apps)
```
{
      "provider":"facebook",
      "access_token":"",
      "access_token_secret":""
}
```
screenshot 2019-03-02 at 01 29 34

**Relevant Pivotal Tracker stories**
[#164047062](https://www.pivotaltracker.com/story/show/164047062)

### CODE REVIEW

File: authors/apps/authentication/migrations/0001_initial.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Generated by Django 2.1.7 on 2019-03-07 10:46
```
Comment (JoanNgatia) [inline authors/apps/authentication/migrations/0001_initial.py]: @kelvinndmo Remove this migrations file from the PR.

File: authors/settings.py [hunk 2]
```diff
@@ -147,5 +153,55 @@
...
+]
+
+
+
+SOCIAL_AUTH_GOOGLE_OAUTH2_KEY = os.environ['SOCIAL_AUTH_GOOGLE_OAUTH2_KEY']
+SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET = os.environ['SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET']
+#Scope
+SOCIAL_AUTH_GOOGLE_OAUTH_SCOPE = ['email', 'username']
+
+SOCIAL_AUTH_TWITTER_KEY = os.environ['SOCIAL_AUTH_TWITTER_KEY']
```
Comment (JoanNgatia) [inline authors/settings.py:168]: Add these env configs to the env.example file in case someone wants to run the stuff locally

**Review by JoanNgatia — CHANGES_REQUESTED** [review]

File: authors/apps/authentication/migrations/0001_initial.py (hunk 1 above)
Comment (kelvinndmo) [reply authors/apps/authentication/migrations/0001_initial.py]: Migration file removed. Thanks for picking out the error

File: authors/settings.py (hunk 2 above)
Comment (kelvinndmo) [reply authors/settings.py:168]: @JoanNgatia added as advised.

File: .env.example [hunk 3]
```diff
@@ -1,7 +1,27 @@
...
-export SENDGRID_API_KEY='API_KEY here' #This is the sendgrid api key
-export EMAIL_FROM="test@test.com" # This is the email that is sending the email
-export URL="http://127.0.0.1:8000/" #The base url where the frontend app is hosted
\ No newline at end of file
+wexport DB_NAME=olympians
+export DB_USER=novak
+export DB_PASSWORD=novak254
+export HOST=''
+
+export SOCIAL_AUTH_GOOGLE_OAUTH2_KEY='65389949271-kspefjrqipki3svnec3dqcsmh92rdgt1.apps.googleusercontent.com'
```
Comment (JoanNgatia) [inline .env.example]: @kelvinndmo For the secret values, you can keep them hidden and replace these values with something like `your google auth key`

File: authors/apps/authentication/views.py [hunk 4]
```diff
@@ -84,6 +88,68 @@ def update(self, request, *args, **kwargs):
...
+            print(user)
+        except BaseException as error:
+            # you cannot associate a social account with more than one user
+
+            return Response({"errors":str(error)},status=status.HTTP_400_BAD_REQUEST
+            )
+        
+        if user and user.is_active:
+            serializer = UserSerializer(user)
+            # authenticated_user_created = user.social.auth.get(provider=provider)
```
Comment (JoanNgatia) [inline authors/apps/authentication/views.py]: Remove the commented out code

**Review by JoanNgatia — CHANGES_REQUESTED** [review]

File: authors/apps/authentication/views.py (hunk 4 above)
Comment (kelvinndmo) [reply authors/apps/authentication/views.py]: Done @JoanNgatia .

File: .env.example (hunk 3 above)
Comment (kelvinndmo) [reply .env.example]: updated @JoanNgatia .Thanks for noting.

**Review by JoanNgatia — APPROVED** [review]

**Pull request merged**